File: meldingsvarsler-fasit-forslag/src/main/java/no/sonat/meldingsvarsler/facebook/FacebookAbonnent.java
package no.sonat.meldingsvarsler.facebook;

import no.sonat.meldingsvarsler.abonnent.Abonnent;
import no.sonat.meldingsvarsler.Melding;

public class FacebookAbonnent implements Abonnent {
    private final String navn;
    private final String brukernavn;

    public FacebookAbonnent(String navn, String brukernavn) {
        this.brukernavn = brukernavn;
        this.navn = navn;
    }
    @Override
    public String navn() {
        return navn;
    }

    @Override
    public Boolean erAktiv() {
        return true;
    }

    @Override
    public boolean abonnererPaa(Melding melding) {
        return melding instanceof FacebookMelding;
    }

    public String brukernavn() {
        return brukernavn;
    }
}

File: meldingsvarsler-fasit-forslag/App/FacebookAbonnent.cs
namespace App {
    public class FacebookAbonnent : IAbonnent
    {
        public FacebookAbonnent(string navn, string brukernavn)
        {
            Brukernavn = brukernavn;
            Navn = navn;
        }

        public string Navn { get; }

        public bool ErAktiv => true;

        public string Brukernavn { get; }

        public bool AbonnererPaa(IMelding melding)
        {
            return melding is FacebookMelding;
        }
    }
}